### PR TITLE
Feature/apb province municipality select filter

### PR DESCRIPTION
**Pull request by Dachamelemon**

### CODE REVIEW

**Comment by Dachamelemon** [comment]
I tried to fix conflicts before merging but i seem to have made a mistake somewhere you can ignore this prequest

File: app/components/administrative-unit-select.hbs [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <div class={{if @error "ember-power-select--error"}} ...attributes>
   <PowerSelect
     @disabled={{if @disabled true false}}
-    @allowClear={{@allowClear}}
+    @allowClear={{not @selectedProvince}}
```
Comment (claire-lovisa) [inline app/components/administrative-unit-select.hbs:4]: I'm not sure we should enforce that we can clear this component only if there is a selected province. It forces the user of the component to use it together with the province component, which might not be relevant for all the use cases. I think that the responsibility to define whether we should allow the clearing is in the hands of the template/controller calling the component instead

File: app/components/administrative-unit-select.js [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
 import { action } from '@ember/object';
 export default class AdministrativeUnitSelectComponent extends Component {
   @service store;
   @tracked loadedRecord;
 
-  constructor() {
-    super(...arguments);
-    this.loadRecord.perform();
-  }
+  munipalities = trackedTask(this, this.loadAdministrativeUnitsTask, () => [
```
Comment (claire-lovisa) [inline app/components/administrative-unit-select.js:13]: Typo in `munipalities` :grin:

File: app/components/province-organization-select.hbs [hunk 3]
```diff
@@ -1,16 +1,17 @@
 <div class={{if @error "ember-power-select--error"}}>
   <PowerSelect
     @disabled={{if @disabled true false}}
-    @allowClear={{true}}
+    @allowClear={{not @selectedMunicipality}}
```
Comment (claire-lovisa) [inline app/components/province-organization-select.hbs:4]: Same as for the admin unit component, I think the component should not handle this but the caller should instead

File: app/templates/administrative-units/new.hbs [hunk 4]
```diff
@@ -163,7 +163,8 @@
               <Item @labelFor="purpose">
                 <:label>Doel</:label>
                 <:content>
-                  <RichTextEditor
+                  <AuTextarea
```
Comment (claire-lovisa) [inline app/templates/administrative-units/new.hbs:166]: I guess it's an error due to the merge conflict, but we do want to use the `RichTextEditor` component here

File: app/templates/administrative-units/administrative-unit/core-data/edit.hbs [hunk 5]
```diff
@@ -187,10 +187,9 @@
                   <Item @labelFor="purpose">
                     <:label>Doel</:label>
                     <:content>
-                      <RichTextEditor
-                        @onChange={{fn
-                          (mut @model.administrativeUnit.purpose)
-                        }}
+                      <AuTextarea
```
Comment (claire-lovisa) [inline app/templates/administrative-units/administrative-unit/core-data/edit.hbs:190]: I guess it's an error due to the merge conflict, but we do want to use the `RichTextEditor` component here

**Review by claire-lovisa — CHANGES_REQUESTED** [review]
We are also entirely missing the functionality in the `Related organizations` tab of APBs, they can also be edited there, can you add it ? :)

File: app/templates/administrative-units/administrative-unit/core-data/edit.hbs [hunk 6]
```diff
@@ -188,9 +188,8 @@
                     <:label>Doel</:label>
                     <:content>
                       <RichTextEditor
-                        @onChange={{fn
-                          (mut @model.administrativeUnit.purpose)
-                        }}
+                        @width="block"
```
[the remark this answers is not in the document]
Comment (claire-lovisa) [reply app/templates/administrative-units/administrative-unit/core-data/edit.hbs:191]: Still has to be rollbacked :)

File: app/templates/administrative-units/new.hbs [hunk 7]
```diff
@@ -164,6 +164,7 @@
                 <:label>Doel</:label>
                 <:content>
                   <RichTextEditor
+                    @width="block"
```
[the remark this answers is not in the document]
Comment (claire-lovisa) [reply app/templates/administrative-units/new.hbs:167]: same !

**Review by claire-lovisa — CHANGES_REQUESTED** [review]
There are still some behaviours that are broken / changed from the default behaviour of the feature:

- Bestuurseenheden page:
  - broke the auto-select of a province when a gemeente is picked up
  - allowClear on the province should be false when a muncipality is selected
  
Core data, edit page:
  - allowClear on the province should be false when a muncipality is selected
  
Related organizations, edit page:
  - autoselect of a province when a gemeente is picked up doesn't work
  - when a municipality is picked up, we can still chose from all the provinces instead of having only one choice
  - allowClear on the province should be false when a muncipality is selected

**Comment by claire-lovisa** [comment]
I'm only seeing it now but this behavior should also be added to the `new` route, when a user created a new administrative unit.

**Review by claire-lovisa — APPROVED** [review]
Thank you :smile:

**Pull request merged**